apps/native/app/src/graphql/client.ts

```diff
@@ -106,11 +107,16 @@ const getAndRefreshToken = async () => {
   if (isTokenPerhapsExpired) {
     // get a new token to be safe
     await refresh()
+    authorizeResult = authStore.getState().authorizeResult
   } else if (isTokenCloseToExpiring) {
     // expires in less than 60 seconds, so refresh in the background
-    refresh().catch((err) => {
-      console.error('Failed to refresh token in the background', err)
-    })
+    refresh()
+      .then(() => {
+        authorizeResult = authStore.getState().authorizeResult
```

Comment:
This is not needed as this happens later in the background. After this function has returned the old (still valid) access token.